src/rgw/rgw_object_lock.h

```diff
@@ -106,7 +106,10 @@ class RGWObjectLock
   }
 
   bool retention_period_valid() const {
-    return (get_years() > 0 || get_days() > 0);
+    // DefaultRetention requires either Days or Years.
+    // You can't specify both at the same time.
+    // see https://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTObjectLockConfiguration.html
+    return (get_years() > 0) != (get_days() > 0);
```

Comment:
nit, no need to put parentheses around the statement after `return`.

Reply:
thanks, there is a warning if we remove those parentheses. 

``` 
/home/liuchang/WorkSpace/ceph/src/rgw/rgw_object_lock.h:112:24: warning: suggest parentheses around comparison in operand of ‘!=’ [-Wparentheses]
     return get_years() > 0 != get_days() > 0;
            ~~~~~~~~~~~~^~~
```